ui/lib/core/addon/components/list-view.js

```diff
@@ -3,20 +3,45 @@ import { computed } from '@ember/object';
 import { pluralize } from 'ember-inflector';
 import layout from '../templates/components/list-view';
 
+/**
+ * @module ListView
+ * `ListView` components are used in conjuction with `ListItem` for rendering a list.
+ *
+ * @example
+ * ```js
+ * <ListView @items={{model}} @itemNoun="role" @paginationRouteName="scope.roles" as |list|>
+ *   {{#if list.empty}}
+ *     <list.empty @title="No roles here" />
+ *   {{else}}
+ *     <div>
+ *       {{list.item.id}}
+ *     </div>
+ *   {{/if}}
+ * </ListView>
+ * ```
+ *
+ * @param items=null {Array} - An array of items to render as a list
+ * @param [itemNoun=null {String}] - A noun to use in the empty state of message and title.
+ * @param [message=null {String}] - The message to display within the banner.
+ * @yields Object with `item` that is the current item in the loop.
```

Comment:
ooh how does jsdom2md handle @yields? does it make it into the story file? We have a custom template for md and I'm not sure it includes these. If it's just for actually documenting in the code here that makes sense too, but I don't think it would make it to the md file.

Reply:
Yeah this was for documentation, but I think it could make sense to pull into a separate table maybe? https://github.com/hashicorp/vault/pull/7129/files/786add6b8b28b4cd84706f73609cf1d2c609b902#diff-db4113d4519f04b3176ff85e5de7e1b4R11 is the param table. Maybe we should look at how ember-cli-addon-docs does it's api doc too because it's 👌 https://ember-learn.github.io/ember-cli-addon-docs/docs/api/components/docs-demo